pkg/chunk/elastic/es_index_client.go

```diff
@@ -0,0 +1,316 @@
+package elastic
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"github.com/cortexproject/cortex/pkg/chunk"
+	chunk_util "github.com/cortexproject/cortex/pkg/chunk/util"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/instrument"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"reflect"
+
+	"fmt"
+	"github.com/olivere/elastic"
+)
+
+const (
+	null = string('\xff')
+)
+
+var esRequestDuration = instrument.NewHistogramCollector(prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Name: "es_request_duration_seconds",
+	Help: "Time spent doing ElasticSearch requests.",
+
+	// from 0us to 10s. TODO: Confirm that this is the case for ApplicationAutoScaling.
```

Comment:
Please revist this comment. It appears to be a copy & paste where some details don't apply to ES